lib/watir/elements/element.rb

```diff
@@ -614,12 +657,14 @@ def assert_is_element(obj)
       end
     end
 
-    def element_call
+    def element_call(exist_check = :wait_for_exists)
+      Wait.timer ||= Wait::Timer.new.tap { |t| t.end_time = ::Time.now + Watir.default_timeout }
```

Comment:
The logic is that the `#wait_for_present` & `#wait_for_exist` methods are only ever accessed by the `#element_call` method. We want to make sure that in the process of verifying parent exists, then parent present, then child exists, then child present, etc, that we do not compound the timeouts. So we create a `Timer` instance, set it as the class variable in Wait and that is what gets used for all calls until the timer expires. We use `||=` such that it only will ever override a `nil` value in case `#element_call` is nested somewhere.